src/psyclone/tests/psyGen_test.py

```diff
@@ -663,6 +664,96 @@ def test_codedkern_lower_to_language_level(monkeypatch):
     assert csymbol.name == "testkern_mod"
 
 
+def test_codedkern__rename_psyir():
```

Comment:
If it doesn't result in massive duplication, please could you break this test into three: structure_type with unsupported routine decln, structure_type with supported routine decln and unsupported structure declaration.

Reply:
Done. I duplicated the "pre-renaming" checks but could leave them in only one test if you wish.

Reply:
Thanks.